consensus/src/network_tests.rs

```diff
@@ -114,13 +126,37 @@ impl NetworkPlayground {
                 // delivery, we'd have to spawn the sending behaviour on a
                 // separate task, which is inconvenient.
                 PeerManagerRequest::SendRpc(dst, outbound_req) => {
+                    let dst_twin_ids = author_to_twin_ids.read().unwrap().get_twin_ids(dst);
+
+                    // Initializing with a dummy value. This variable will only be
+                    // used if drop_rpc is false, in which case it will be updated to
+                    // a proper value
+                    let mut dst_twin_id = src_twin_id.clone();
+                    let mut drop_rpc = true;
+
+                    // We send RPC message to any of the twins that's not blocked.
+                    // If all twins are blocked, then drop_rpc will be true and
+                    // the message will not be sent at all.
+                    for each_dst_twin_id in dst_twin_ids.iter() {
```

Comment:
this whole section can be simplified as
```
let dst = match dst_twins_ids.iter().find(|dst| !drop_config.read().unwrap().is_message_dropped(..)) {
  Some(id) => id,
  None => continue,
};
```

Reply:
That's indeed much more elegant, thanks. Done.